include/wil/chrono.h

```diff
@@ -0,0 +1,374 @@
+            {
+                return from_unix_epoch(t.time_since_epoch());
+            }
+
+#ifndef _CRT_NO_TIME_T
+            WI_NODISCARD static __WI_LIBCPP_CONSTEXPR_AFTER_CXX11 ::time_t to_time_t(const time_point& t) WI_NOEXCEPT
+            {
+                return to_crt_time<::time_t>(t);
+            }
+
+            WI_NODISCARD static __WI_LIBCPP_CONSTEXPR_AFTER_CXX11 time_point from_time_t(::time_t t) WI_NOEXCEPT
+            {
+                return from_crt_time(t);
+            }
+#endif // !defined(_CRT_NO_TIME_T)
+
+            WI_NODISCARD static __WI_LIBCPP_CONSTEXPR_AFTER_CXX11 ::__time32_t to_time32_t(const time_point& t) WI_NOEXCEPT
+            {
+                return to_crt_time<::__time32_t>(t);
+            }
+
+            WI_NODISCARD static __WI_LIBCPP_CONSTEXPR_AFTER_CXX11 time_point from_time32_t(::__time32_t t) WI_NOEXCEPT
+            {
+                return from_crt_time(t);
+            }
+
+            WI_NODISCARD static __WI_LIBCPP_CONSTEXPR_AFTER_CXX11 ::__time64_t to_time64_t(const time_point& t) WI_NOEXCEPT
+            {
+                return to_crt_time<::__time64_t>(t);
+            }
+
+            WI_NODISCARD static __WI_LIBCPP_CONSTEXPR_AFTER_CXX11 time_point from_time64_t(::__time64_t t) WI_NOEXCEPT
+            {
+                return from_crt_time(t);
+            }
+        };
+
+#if __WI_LIBCPP_STD_VER <= 11
+        template<typename get_system_time_fn_t, get_system_time_fn_t get_system_time_fn, typename base_clock_t>
+        typename system_time_clock_impl<get_system_time_fn_t, get_system_time_fn, base_clock_t>::time_point const system_time_clock_impl<get_system_time_fn_t, get_system_time_fn, base_clock_t>::unix_epoch{duration{116444736000000000LL}};
+#endif
+    }
+
+    struct system_time_clock : details::system_time_clock_impl<decltype(&::GetSystemTimeAsFileTime), ::GetSystemTimeAsFileTime, system_time_clock> {};
+
+#if _WIN32_WINNT >= _WIN32_WINNT_WIN8
+    struct precise_system_time_clock : details::system_time_clock_impl<decltype(&::GetSystemTimePreciseAsFileTime), ::GetSystemTimePreciseAsFileTime, system_time_clock> {};
+    using high_precision_system_time_clock = precise_system_time_clock;
+#else // _WIN32_WINNT < _WIN32_WINNT_WIN8
+    using high_precision_system_time_clock = system_time_clock;
+#endif // _WIN32_WINNT < _WIN32_WINNT_WIN8
+#pragma endregion
+
+#pragma region std::chrono wrappers for Query[Unbiased]InterruptTime[Precise]
+    namespace details
+    {
+        template<typename query_interrupt_time_fn_t, query_interrupt_time_fn_t query_interrupt_time_fn, typename base_clock_t>
+        struct interrupt_time_clock_impl
+        {
+            using rep = LONGLONG;
+            using period = std::ratio_multiply<std::hecto, std::nano>;
+            using duration = std::chrono::duration<rep, period>;
+            using time_point = std::chrono::time_point<base_clock_t, duration>;
+            static constexpr bool is_steady = true;

```

Comment:
[The phrasing from the documentation](https://docs.microsoft.com/en-us/windows/win32/api/realtimeapiset/nf-realtimeapiset-queryunbiasedinterrupttime):

> The unbiased interrupt-time count does not include time the system spends in sleep or hibernation.

Makes it sound like this should be `false`. At least the standard makes it sound like it should be:

> `true` if `t1 <= t2` is always `true` and the time between clock ticks is constant, otherwise `false`.